components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+export interface Props{
+
+    /**
+     * Permissions enough to pass the gate (binary OR)
+     */
+    permissions: string[],
+
+    /**
+     * Has permission
+     */
+    hasPermission?: boolean,
```

Comment:
This prop is marked as `isRequired` in the original file. Is there a reason it was changed to optional here?

Reply:
The reason is that ts complains about this prop because it's added via `connect` in `/any_team_permission_gate_index`. I tried a few ways of typing it but didn't manage to fix this issue. Maybe you know how to deal with that?

Reply:
Where is it complaining? Usually the TS compiler is able to detect props added by `redux`.
Can you post the error you're getting?

Reply:
![image](https://user-images.githubusercontent.com/1397979/95212065-fc86b280-07f5-11eb-9303-68a277701ca9.png)
Here is an example of `npm run check-types` with `hasPermission` required

Reply:
Ahh alright, so the test files do not import the redux-connected component. We expect to mock out those properties ourselves.

So in the case of this test, if `hasPermission` is missing it would evaluate to `false` within the component. Meaning you'd want to add `hasPermission={false}` to that particular test.

Reply:
![image](https://user-images.githubusercontent.com/1397979/95223311-1a5a1480-0802-11eb-8bd6-e7f1af6c8859.png)
It's not only the test problem. Here is an example of another component with an error, and you can see, that it imports the connected component. Will try to find the root cause of the issue.

Reply:
I hit the same issue in another PR, in that case, I thought it is because the parent component is in js 🤷

Reply:
@devinbinnie @sudheerDev fixed this issue by setting explicit types in the `connect`

Reply:
`any_team_permission_gate.tsx` was converted to `tsx` in other MR, so I reverted my changes from those files